azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatServiceListener.kt

```diff
@@ -40,6 +45,41 @@ internal class ChatServiceListener(
                 onMessagesPageModelReceived(messagesPageModel = it, dispatch = dispatch)
             }
         }
+
+        coroutineScope.launch {
+            chatService.getChatEventSharedFlow().collect {
+                when (it.eventType) {
```

Comment:
Instead of casting, can we invert to make cast-safe.

i.e. 
when (it.infoModel) {
  is MessageInfoModel -> handleMessageInfoModel(it.infoModel, it.eventType)
}

To allow type-safe casting.